### PR TITLE
[ci] Update NVIDIA driver to v535 and CUDA to v12.2.1

### PR DESCRIPTION
**Pull request by antiagainst**
This commits bumps NVIDIA drivers to v535.86.05 for both CUDA and Vulkan. It also updates CUDA SDK to v12.2.1.

### CODE REVIEW

File: build_tools/docker/dockerfiles/nvidia.Dockerfile [hunk 1]
```diff
@@ -11,12 +11,12 @@
 # We use .deb files that we host because we have to pin the version and packages
 # routinely dissapear from the Ubuntu apt repositories. The versions need to be
 # compatible with the host driver (usually <= host driver version).
-ARG NVIDIA_GL_DEB="libnvidia-gl-530_530.41.03-0ubuntu0.20.04.2_amd64.deb"
-ARG NVIDIA_COMPUTE_DEB="libnvidia-compute-530_530.41.03-0ubuntu0.20.04.2_amd64.deb"
-ARG NVIDIA_COMMON_DEB="libnvidia-common-530_530.41.03-0ubuntu0.20.04.2_all.deb"
```
Comment (pzread) [inline build_tools/docker/dockerfiles/nvidia.Dockerfile:16]: The NVIDIA driver on host is still 530.41

```
NVIDIA-SMI 530.41.03              Driver Version: 530.41.03    CUDA Version: 12.1
```

Do we need to upgrade them?

Comment (antiagainst) [reply build_tools/docker/dockerfiles/nvidia.Dockerfile:16]: I think it's fine. CUDA 12.2.x requires >=525.60.13.

Comment (antiagainst) [reply build_tools/docker/dockerfiles/nvidia.Dockerfile:16]: Nah. It turns out we also need to update the base VM: https://github.com/openxla/iree/actions/runs/5981136086/job/16228796396?pr=14833#step:7:164 :(

@pzread could you help with that? Thanks!

Comment (pzread) [reply build_tools/docker/dockerfiles/nvidia.Dockerfile:16]: I'll do it early this week

Comment (antiagainst) [reply build_tools/docker/dockerfiles/nvidia.Dockerfile:16]: Thanks @pzread!

Comment (pzread) [reply build_tools/docker/dockerfiles/nvidia.Dockerfile:16]: I set up testing runners with the new NVIDIA driver. The tests can run now (but some of them are failing): https://github.com/openxla/iree/actions/runs/5981136086/job/16283843823?pr=14833

FYI, I put these lines in the PR description so CI will run with testing runners and only run `test_gpu` job to save time. We will need to remove it before submit for a complete CI run.

```
runner-env: testing
ci-exactly: build_all, test_gpu
```

Comment (pzread) [reply build_tools/docker/dockerfiles/nvidia.Dockerfile:16]: The nvidia driver on all runners has been upgraded to `535.86.05`

Comment (antiagainst) [reply build_tools/docker/dockerfiles/nvidia.Dockerfile:16]: Nice, thanks @pzread! I'll try it again now.

**Comment by antiagainst** [comment]
Unfortunately this does not fix the tests: https://github.com/openxla/iree/actions/runs/6051489758/job/16423592469?pr=14833#step:7:1118. So I'll revert the test enabling commit and land the infra change only.

**Comment by ScottTodd** [comment]
> Unfortunately this does not fix the tests: https://github.com/openxla/iree/actions/runs/6051489758/job/16423592469?pr=14833#step:7:1118. So I'll revert the test enabling commit and land the infra change only.

Ok. I merged my workaround for the test failures in https://github.com/openxla/iree/pull/14790

**Review by ScottTodd — APPROVED** [review]

File: third_party/nvidia_sdk_download/fetch_cuda_toolkit.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
 import subprocess
 import sys
 
-VERSION = "12.1.1"
+VERSION = "12.2.1"
```
Comment (Groverkss) [inline third_party/nvidia_sdk_download/fetch_cuda_toolkit.py:22]: This change is outside CI btw. This broke my build and took some time to figure out what happened. Maybe announce it outside CI changes too?

**Pull request merged**